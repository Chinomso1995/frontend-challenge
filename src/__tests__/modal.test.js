import React from 'react';
import TestRenderer from 'react-test-renderer';
import FormModal from '../components/modal';

// Mock the useToast hook
jest.mock('@chakra-ui/react', () => ({
  ...jest.requireActual('@chakra-ui/react'),
  useToast: jest.fn(() => ({ // Mock the useToast function
    toast: jest.fn(),
  })),
}));

test('renders the form modal correctly', () => {
  // Render the modal component
  const component = TestRenderer.create(<FormModal />);

  // Capture the snapshot of the rendered output
  const tree = component.toJSON();

  // Assert the snapshot
  expect(tree).toMatchSnapshot();
});
