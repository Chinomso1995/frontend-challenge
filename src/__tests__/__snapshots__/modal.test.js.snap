// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the form modal correctly 1`] = `
<button
  className="chakra-button css-i0gsjp"
  type="button"
>
  Anadir tarea
</button>
`;
